### ISSUE TITLE
`IllegalStateException` in `Desugar` for const with user defined type

### ISSUE THREAD

**Issue by rdulmina** [ballerina-platform/ballerina-lang#35510]
**Description:**

The below sample causes `IllegalStateException` in `Desugar`

```bal
type Duration decimal;
const decimal SECOND = 1;
const Duration HOUR = 60*SECOND;
```
<details>
  <summary>Stack trace.(Click to expand)</summary>

```
[2022-03-21 12:18:56,109] SEVERE {b7a.log.crash} - null 
java.lang.IllegalStateException
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:8096)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangConstant.accept(BLangConstant.java:112)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8343)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8394)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.visit(Desugar.java:781)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.rewrite(Desugar.java:8343)
        at org.wso2.ballerinalang.compiler.desugar.Desugar.perform(Desugar.java:454)
        at io.ballerina.projects.internal.CompilerPhaseRunner.desugar(CompilerPhaseRunner.java:216)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performBirGenPhases(CompilerPhaseRunner.java:152)
        at io.ballerina.projects.ModuleContext.generateCodeInternal(ModuleContext.java:453)
        at io.ballerina.projects.ModuleCompilationState$4.generatePlatformSpecificCode(ModuleCompilationState.java:132)
        at io.ballerina.projects.ModuleContext.generatePlatformSpecificCode(ModuleContext.java:390)
        at io.ballerina.projects.JBallerinaBackend.performCodeGen(JBallerinaBackend.java:159)
        at io.ballerina.projects.JBallerinaBackend.<init>(JBallerinaBackend.java:137)
        at io.ballerina.projects.JBallerinaBackend.lambda$from$0(JBallerinaBackend.java:113)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1133)
        at io.ballerina.projects.PackageCompilation.getCompilerBackend(PackageCompilation.java:182)
        at io.ballerina.projects.JBallerinaBackend.from(JBallerinaBackend.java:112)
        at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:93)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.RunCommand.execute(RunCommand.java:203)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:51)
```
</details>

**Affected Versions:**
2201.0.2

**Comment by pcnfernando** [comment]
Typeref cases should be added to https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java#L424

**Comment by rdulmina** [comment]
Fixed with #37026
